src/operator/nn/softmax.cc

```diff
@@ -12,6 +12,27 @@ namespace op {
 DMLC_REGISTER_PARAMETER(SoftmaxParam);
 
 MXNET_OPERATOR_REGISTER_UNARY(softmax)
+.describe(R"code(Applies the Softmax function on the input array re-scaling 
+the elements such that they lie in the range (0,1) and the elements along 
+the given axis sum up to 1.
```

Comment:
changed to 
**Applies the softmax function. The elements are re-scaled such that they lie in the range (0,1) and the elements along the given axis sum up to 1.**

true, it sounds like its doing linear transformation. but isn't cross-entropy along with softmax that makes it non-linear?
regarding the open-end, for the case you described it does return 1 but for everything else it does not 1. I will leave it open ended for now following other documentation.

this was borrowed from [pytorch](https://nn.readthedocs.io/en/rtd/transfer/#softmax).